src/user_lists/serializers.py

```diff
@@ -15,6 +22,137 @@ class Meta:
             "updated_date",
             "created_by",
             "updated_by",
+            "items_count",
         ]
         read_only_fields = ["id", "created_date", "updated_date", "created_by", "updated_by"]
 
+    def get_items_count(self, obj):
+        if hasattr(obj, '_prefetched_objects_cache') and 'items' in obj._prefetched_objects_cache:
+            return len([item for item in obj._prefetched_objects_cache['items'] if not item.is_removed])
+        return obj.items.filter(is_removed=False).count()
+
+
+class ListItemSerializer(DefaultAuthenticatedSerializer):
+    unified_document = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ListItem
+        fields = ["id", "parent_list", "unified_document", "created_date", "created_by"]
+        read_only_fields = ["id", "created_date", "created_by"]
+        extra_kwargs = {
+            'unified_document': {'required': True}
+        }
+
+    def get_fields(self):
+        fields = super().get_fields()
+        if self.context.get('request') and self.context['request'].method in ['POST', 'PUT', 'PATCH']:
+            fields['unified_document'] = serializers.PrimaryKeyRelatedField(
+                queryset=ResearchhubUnifiedDocument.objects.filter(is_removed=False), 
+                required=True
+            )
+        return fields
+
+    def get_unified_document(self, obj):
```

Comment:
`get_unified_document` should return unified document not FeedEntry. If we don't want it to return unified document let's change the name